lib/ZwaveClient.ts

```diff
@@ -0,0 +1,3276 @@
+	isFrequentListening?: FLiRS
+	isRouting?: boolean
+	keepAwake?: boolean
+	deviceClass?: Z2MDeviceClass
+	neighbors?: number[]
+	loc?: string
+	name?: string
+	hassDevices?: { [key: string]: HassDevice }
+	deviceId?: string
+	hexId?: string
+	values?: { [key: string]: Z2MValueId }
+	groups?: Z2MNodeGroups[]
+	ready: boolean
+	available: boolean
+	failed: boolean
+	lastActive?: number
+	dbLink?: string
+	maxDataRate?: DataRate
+	interviewStage?: keyof typeof InterviewStage
+	status?: keyof typeof NodeStatus
+	inited: boolean
+	healProgress?: string | undefined
+}
+
+export type ZwaveConfig = {
+	port: string
+	networkKey: string
+	serverEnabled: boolean
+	serverPort: number
+	logEnabled: boolean
+	logLevel: LogManager.LogLevel
+	commandsTimeout: number
+	enableStatistics: boolean
+	disclaimerVersion: number
+	options: ZWaveOptions
+	healNetwork: boolean
+	healHour: number
+	logToFile: boolean
+	nodeFilter: string[]
+}
+
+export type Z2MDriverInfo = {
+	uptime?: number
+	lastUpdate?: number
+	status?: ZwaveClientStatus
+	cntStatus?: string
+	appVersion?: string
+	zwaveVersion?: string
+	serverVersion?: string
+	homeid?: number
+	name?: string
+	controllerId?: number
+	newConfigVersion?: string | undefined
+}
+
+export enum ZwaveClientStatus {
+	CONNECTED = 'connected',
+	DRIVER_READY = 'driver ready',
+	SCAN_DONE = 'scan done',
+	DRIVER_FAILED = 'driver failed',
+	CLOSED = 'closed',
+}
+
+export enum EventSource {
+	DRIVER = 'driver',
+	CONTROLLER = 'controller',
+	NODE = 'node',
+}
+
+export interface ICallApiResult {
+	message: any
+	args?: any
+	success?: boolean
+	result?: any
+}
+
+declare interface ZwaveClient {
+	on(event: 'nodeStatus', listener: (node: Z2MNode) => void): this
+	on(
+		event: 'event',
+		listener: (source: EventSource, eventName: string, ...args: any) => void
+	): this
+	on(event: 'scanComplete', listener: () => void): this
+	on(
+		event: 'notification',
+		listener: (node: Z2MNode, valueId: Z2MValueId, data: any) => void
+	): this
+	on(event: 'nodeRemoved', listener: (node: Z2MNode) => void): this
+	on(
+		event: 'valueChanged',
+		listener: (valueId: Z2MValueId, node: Z2MNode) => void
+	): this
+	on(
+		event: 'valueWritten',
+		listener: (valueId: Z2MValueId, value: any) => void
+	): this
+}
+
+class ZwaveClient extends EventEmitter {
+	cfg: ZwaveConfig
+	socket: SocketServer
+	closed: boolean
+	driverReady: boolean
+	scenes: Z2MScene[]
+	nodes: Map<number, Z2MNode>
+	storeNodes: Record<number, Partial<Z2MNode>>
+	devices: Record<string, Partial<Z2MNode>>
+	driverInfo: Z2MDriverInfo
+	status: ZwaveClientStatus
+	error: boolean | string
+	scanComplete: boolean
+	cntStatus: string
+	connected: boolean
+	lastUpdate: number
+
+	driver: Driver
+
+	server: ZwavejsServer
+	statelessTimeouts: Record<string, NodeJS.Timeout>
+	commandsTimeout: NodeJS.Timeout
+	reconnectTimeout: NodeJS.Timeout
+	healTimeout: NodeJS.Timeout
+	updatesCheckTimeout: NodeJS.Timeout
+	pollIntervals: Record<string, NodeJS.Timeout>
```

Comment:
class properties should have `public`, `protected` or `private` modifiers. This doesn't really do anything during runtime, but clearly states the intent of visibility.
Same goes for all methods.

Reply:
Yep I know, as a first step I focused on make the things work then I planned to gradually fix this things